### PR TITLE
Client ws d1

### PR DESCRIPTION
**Pull request by marlukyanova**

### CODE REVIEW

**Review by Gabss405 — APPROVED** [review]

File: client/src/containers/Avatar/Avatar.tsx [hunk 1]
```diff
@@ -1,31 +1,50 @@
...
+    socketRef.current = SocketIOCLient(SOCKET_SERVER_URL, {
+      query: {roomId}
+    });
+    props.setSocket(socketRef);
+  }, [roomId]);
 
-  // console.log('created socket', socketRef);
+  //console.log('created socket', props.socket);
 
+  function handleClick(): void {
```
Comment (joedavies25) [inline client/src/containers/Avatar/Avatar.tsx:34]: more descriptive naming

File: client/src/containers/Avatar/Avatar.tsx [hunk 2]
```diff
@@ -1,31 +1,50 @@
...
 
-const Avatar: React.FC<RouteComponentProps<MatchParams>> = (props) => {
-  // const socketRef = useRef();
-  const roomId = props.match.params.roomId;
+import './styles/Avatar.scss';
 
-  const socket = SocketIOCLient(SOCKET_SERVER_URL, {
-    query: { roomId },
-  });
+type Props = {
```
Comment (hectorkd) [inline client/src/containers/Avatar/Avatar.tsx:9]: Rename prop to specify what page it's for. Is type better for this than interface?

File: client/src/containers/Lobby/Lobby.tsx [hunk 3]
```diff
@@ -1,12 +1,53 @@
...
+    //TODO: store paragraph
+    // props.setText(text);
+  });
+
+  props.socket.current.on('players', (roomId: string, players: Player[]) => {
+    //console.log(players);
+    //TODO: show players on a page
+  })
+
+  function handleClick(): void {
```
Comment (joedavies25) [inline client/src/containers/Lobby/Lobby.tsx:34]: more descriptive naming

File: client/src/containers/Lobby/Lobby.tsx [hunk 4]
```diff
@@ -1,12 +1,53 @@
-import React from 'react';
+import React, {ReactNode} from 'react';
+import { useParams, useHistory } from 'react-router-dom';
+
+import { Player } from '../../interfaces/Player';
 import PlayersList from '../../components/PlayerList /PlayerList';
+
 import './styles/Lobby.scss';
 
+type Props = {
```
Comment (hectorkd) [inline client/src/containers/Lobby/Lobby.tsx:9]: Rename prop

File: client/src/containers/Race/Race.tsx [hunk 5]
```diff
@@ -1,8 +1,92 @@
-import React from 'react';
+import React, { ReactNode } from 'react';
+import useTypingGame from '../../useTypingGame';
+
 import './styles/Race.scss';
 
-const Race: React.FC = () => {
-  return <div></div>;
+type Props = {
```
Comment (hectorkd) [inline client/src/containers/Race/Race.tsx:6]: Rename prop

**Review by joedavies25 — APPROVED** [review]

**Review by hectorkd — APPROVED** [review]

**Pull request merged**